official/recommendation/data_download.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Download and extract the movielens dataset from grouplens website.
+
+Download the dataset, and perform data-preprocessing to convert the raw dataset
+into csv file to be used in model training and evaluation.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import collections
+import os
+import sys
+import time
+import urllib
+import zipfile
+
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+import numpy as np
+import pandas as pd
+
+USER_COLUMN = 'user_id'
+ITEM_COLUMN = 'item_id'
+NUMBER_NEGATIVES = 100
+
+TRAIN_RATINGS_FILENAME = 'train-ratings.csv'
+TEST_RATINGS_FILENAME = 'test-ratings.csv'
+TEST_NEG_FILENAME = 'test-negative.csv'
+
+DATA_URL = 'http://files.grouplens.org/datasets/movielens/'
+
+RatingData = collections.namedtuple(
+    'RatingData', ['items', 'users', 'ratings', 'min_date', 'max_date'])
+
+
+def _describe_ratings(ratings):
+  """Describe the rating dataset information.
```

Comment:
Since it does not return anything, probably should say its printing out the dataset info on stdout.